### PR TITLE
Laboratory work #3, Zakirova Karina - 20FPL1 

### PR DESCRIPTION
**Pull request by KarinaZakirova**

### CODE REVIEW

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]
Good start

File: lab_3/get_threegram_frequencies_test.py [hunk 1]
```diff
@@ -25,9 +25,11 @@ def test_calculate_ngram_frequencies_ideal(self):
 And when I am introduced to one
 I wish I thought What Jolly Fun'''.lower()
         text = tokenize_by_sentence(text)
+        print(text)
```
Comment (WhiteJaeger) [inline lab_3/get_threegram_frequencies_test.py:28]: Do not forget to remove.

File: lab_3/main.py [hunk 2]
```diff
@@ -222,7 +337,21 @@ def save(self, name: str) -> int:
...
+            return 1
+        freq = {}
+        profile_as_dict = {}
+        for n_gram_trie in self.tries:
+            freq.update(((''.join(map(self.storage.get_letter_by_id, k)), v)
+                         for k, v in n_gram_trie.n_gram_frequencies.items()))
+        profile_as_dict["freq"] = freq
+        profile_as_dict["n_words"] = self.n_words
+        profile_as_dict["name"] = self.language
+        with open(name, "w", encoding="utf-8") as file:
```
Comment (WhiteJaeger) [inline lab_3/main.py:350]: `file` is the Python built-in. Consider using another name.

File: lab_3/main.py [hunk 3]
```diff
@@ -222,7 +337,21 @@ def save(self, name: str) -> int:
...
+            json_string = json.dumps(profile_as_dict)
```
Comment (WhiteJaeger) [inline lab_3/main.py:351]: `json` has the `dump` method that would take care of writing to the file. So this could be combined with the following one.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_3/get_threegram_frequencies_test.py (hunk 1 above)
Comment (KarinaZakirova) [reply lab_3/get_threegram_frequencies_test.py:28]: okay

File: lab_3/main.py (hunk 2 above)
Comment (KarinaZakirova) [reply lab_3/main.py:350]: done

File: lab_3/main.py (hunk 3 above)
Comment (KarinaZakirova) [reply lab_3/main.py:351]: I see
done

**Review by WhiteJaeger — APPROVED** [review]

**Pull request merged**